app/controllers/api/cloud_networks_controller.rb

```diff
@@ -9,5 +9,23 @@ def delete_resource(type, resource_id, _data = {})
       task_id = cloud_network.delete_cloud_network_queue(User.current_user.userid)
       action_result(true, "Deleting #{cloud_network.name}", :task_id => task_id)
     end
+
+    def options
+      return super unless params.keys.include_any?(%w[id ems_id])
+
+      if params.key?("ems_id")
```

Comment:
I think this might be cleaner as follows:

```ruby
def options
  if params.key?("ems_id")
    options_by_ems_id
  elsif params.key?("id")
    options_by_id
  else
    super
  end
end
```

Then break out these other 2 sections into their own private methods `options_by_ems_id` and `options_by_id`. 

WDYT?

Reply:
(Maybe not for this PR, especially if this pattern was copy/pasted)

Reply:
I think splitting them up is cleaner, I can change it right now if you want it split up

Reply:
Go for it 😊